src/wasm/wasm.cpp

```diff
@@ -923,16 +923,20 @@ void CallRef::finalize() {
   handleUnreachableOperands(this);
   if (isReturn) {
     type = Type::unreachable;
+    return;
   }
   if (target->type == Type::unreachable) {
     type = Type::unreachable;
+    return;
+  }
+  assert(target->type.isRef());
+  if (target->type.getHeapType().isBottom()) {
+    return;
```

Comment:
Should we perhaps set the type to unreachable here? That would avoid the extra line in `wasm-binary`, and seems simpler, but maybe I'm missing a corner case...

Reply:
I tried doing this initially, but it led to extra complications because our validator checks that with very few exceptions, unreachable nodes must have unreachable children. It may make sense to overhaul this in the future so that all instructions that trap on null (e.g. struct.get, struct.set, etc.) all get type `unreachable` when their input has bottom type. For now, the current behavior is the most consistent with other trapping instructions.

Reply:
Ah right... yeah, that makes sense.